saleor/plugins/avatax/plugin.py

```diff
@@ -421,6 +423,63 @@ def get_tax_rate_type_choices(self, previous_value: Any) -> List[TaxType]:
             for tax_code, desc in get_cached_tax_codes_or_fetch(self.config).items()
         ]
 
+    def get_checkout_tax_rate(
+        self,
+        checkout: "Checkout",
+        product: "Product",
+        address: Optional["Address"],
+        checkout_line: "CheckoutLine",
+        discounts: Iterable[DiscountInfo],
+        previous_value: Decimal,
+    ) -> Decimal:
+        return self._get_tax_rate(
+            checkout, previous_value, False, discounts, [checkout_line]
+        )
+
+    def get_order_tax_rate(
+        self,
+        order: "Order",
+        product: "Product",
+        address: Optional["Address"],
+        previous_value: Decimal,
+    ) -> Decimal:
+        return self._get_tax_rate(order, previous_value, True)
+
+    def _get_tax_rate(
+        self,
+        instance: Union["Order", "Checkout"],
+        base_rate: Decimal,
+        is_order: bool,
+        discounts: Optional[Iterable[DiscountInfo]] = None,
+        checkout_lines: Iterable["CheckoutLine"] = [],
+    ):
+        if self._skip_plugin(base_rate):
+            return base_rate
+
+        valid = (
+            _validate_order(instance)  # type: ignore
+            if is_order
+            else _validate_checkout(instance, checkout_lines)  # type: ignore
+        )
+
+        if not valid:
+            return base_rate
+
+        response = (
+            get_order_tax_data(instance, self.config, False)  # type: ignore
```

Comment:
shouldn't we handle `TaxError` here and return `base_rate` if it's raised?

Reply:
I'm not sure if we shouldn't raise an error in this situation. I checked other places where `get_order_tax_data` is used, and in any of these places `TaxError` is handled in the plugin. I also checked, that `TaxError` is handled by methods that called functions that use tax rate methods from plugins.